ansible_collections/arista/avd/roles/eos_designs/python_modules/pool_manager/avdpoolmanager.py

```diff
@@ -0,0 +1,168 @@
+            id_data["id_pools"] = natural_sort(id_data["id_pools"], "dc_name")
+            id_data["id_pools"] = natural_sort(id_data["id_pools"], "fabric_name")
+            id_file.write_text(FILE_HEADER + dump(id_data, Dumper=Dumper), encoding="UTF-8")
+        self.id_pool_data.cache_clear()
+        self.changed_id_files = set()
+        self.id_files = set()
+
+    @lru_cache
+    def id_pool_data(self, id_file: Path) -> dict:
+        if not id_file.exists():
+            # Try to create the dir and file.
+            id_file.parent.mkdir(mode=0o775, parents=True, exist_ok=True)
+            id_file.touch(mode=0o664)
+
+        # Read from file or assign an empty data model.
+        id_data = safe_load(id_file.read_text(encoding="UTF-8")) or {"id_pools": []}
+
+        if not isinstance(id_data, dict) or not isinstance(id_data.get("id_pools"), list):
+            raise AristaAvdError(f"Invalid ID pool manager data when reading {id_file}. Expecting {'id_pools': []}")
+
+        # Since we assigned a new ID, we have to mark the id pool as changed to have it saved once the build is done.
+        self.id_files.add(id_file)
+
+        return id_data
+
+    def get_id(self, shared_utils: SharedUtils) -> int | None:
+        """
+        Returns the ID of one node if found in the pool.
+        Otherwise a new entry is inserted into the pool and the ID is returned.
+
+        The pools are dynamically built and matched on the following device data:
+        - Fabric Name (from 'shared_utils.hostvars.fabric_name')
+        - DC Name (from `shared_utils.dc_name`)
+        - POD Name (from `shared_utils.pod_name`)
+        - Node type (from `shared_utils.type`)
+
+        Each pool will assign first available ID starting from 1.
+
+        Args:
+            shared_utils: Instance of SharedUtils initialized for one device.
+
+        Returns:
+            Node ID
+        """
+        fabric_name = get(shared_utils.hostvars, "fabric_name", required=True)
+        default_id_file = f"{self.output_dir}/data/{fabric_name}-ids.yml"
+        id_file = Path(get(shared_utils.hostvars, "fabric_numbering.id.pools_file", default=default_id_file))
+        id_data = self.id_pool_data(id_file)
+
+        id_pool = None
+        for id_data_id_pool in id_data["id_pools"]:
+            if (
+                id_data_id_pool.get("fabric_name") == fabric_name
+                and id_data_id_pool.get("dc_name") == shared_utils.dc_name
+                and id_data_id_pool.get("pod_name") == shared_utils.pod_name
+                and id_data_id_pool.get("type") == shared_utils.type
+            ):
+                id_pool = id_data_id_pool
+                break
+
+        if not id_pool:
+            # Create a new dict, add to the list of pools.
+            id_pool = {
+                "fabric_name": fabric_name,
+                "dc_name": shared_utils.dc_name,
+                "pod_name": shared_utils.pod_name,
+                "type": shared_utils.type,
```

Comment:
I get why we need to match on the `type` but from the perspective of the campus studio leveraging pyavd, this may result in issues if a user were to re-architect their network from an L2 campus to an L3 campus because of the way Node IDs work in the campus studio.  Basically, for an L2 and L3 campus design in the campus studio we treat leafs and member leafs as different types.  In AVD, in an L2 campus the design is `l2ls` and leafs and member leafs are both `leafs` while in an L3 campus the design is `l3ls-evpn` and leafs are `l3leafs` and member leafs are `l2leafs`.  The solution to this may not require anything with the pool manager and may just involve adding another node type to the `l2ls` avd design.

Reply:
It would require some rework of the campus studio logic, to maybe introduce another type like member-leaf or whatever we decide. We have other things to discuss around this, like simpler data models for such member leafs etc. I am also okay with adding another "layer" in here if we feel for it, but let's talk first.

Reply:
Another option is to also add the `rack` property we have already. It could be exposed as "IDF" in the campus scenario.

Reply:
I will address this after testing.